### PR TITLE
Nim-1.4 compatibility

### PR DESCRIPTION
**Pull request by stefantalpalaru**

### CODE REVIEW

File: beacon_chain/spec/digest.nim [hunk 1]
```diff
@@ -99,6 +99,9 @@ template hash*(x: Eth2Digest): Hash =
   cast[ptr Hash](unsafeAddr x.data[0])[]
 
 func `==`*(a, b: Eth2Digest): bool =
-  # nimcrypto uses a constant-time comparison for all MDigest types which for
-  # Eth2Digest is unnecessary - the type should never hold a secret!
-  equalMem(unsafeAddr a.data[0], unsafeAddr b.data[0], sizeof(a.data))
+  when nimvm:
+    a.data[0] == b.data[0]
```
Comment (arnetheduck) [inline beacon_chain/spec/digest.nim]: comparing the first byte??

Comment (stefantalpalaru) [reply beacon_chain/spec/digest.nim]: Oh, that's an array. Fixed.

**Comment by arnetheduck** [comment]
Hm, that's quite a massive nimyaml update, what does it contain apart from 1.4 compat?

**Comment by stefantalpalaru** [comment]
My own Nim-1.2 compatibility fixes, since upstream gave up on it, and a copy of their test suite that was still implemented as a submodule targetting a commit that's no longer on any branch.

**Comment by arnetheduck** [comment]
ah test suite - was more curious what they changed and why they need 1.4 for it

**Comment by stefantalpalaru** [comment]
See https://github.com/flyx/NimYAML/compare/f8b264f99291a..fcae534

**Pull request merged**